### PR TITLE
Approximate file match ranking #240

### PR DESCRIPTION
**Pull request by JonoYang**

### CODE REVIEW

File: matchcode/models.py [hunk 1]
```diff
@@ -307,22 +315,90 @@ def match(cls, fingerprint, exact_match=False):
...
+                    logger_debug(
+                        cls.__name__,
+                        'match:',
+                        'step_4_matched_resource:',
+                        matched_resource
+                    )
+
+                # Compute size and name distance (similarity)
+                if matched_resource.is_file:
+                    size_distance = 1 - (resource_size / matched_resource.size)
```
Comment (JonoYang) [inline matchcode/models.py:366]: @pombredanne Should this be `1 - (resource_size / matched_resource.size)` or `1 - (matched_resource.size / resource_size)`?

**Review by JonoYang — COMMENTED** [review]
@pombredanne I have a question about creating the size distance

**Comment by pombredanne** [comment]
> I have a question about creating the size distance
For reference, based on our chat, we discussed using the absolute size difference between the queried scanned code and the matched index code as the size distance and use the smallest as a disambiguation factor.

File: matchcode/models.py [hunk 2]
```diff
@@ -307,22 +314,88 @@ def match(cls, fingerprint, exact_match=False):
...
+                    logger_debug(
+                        cls.__name__,
+                        'match:',
+                        'step_4_size_difference:',
+                        size_difference,
+                        'step_4_name_difference:',
+                        name_difference
+                    )
+
+        # Order these from low to high (low being low difference/very similar))
```
Comment (keshav-space) [inline matchcode/models.py:383]: We should be more explicit about the order of ranking.

```suggestion
        # Order these from low to high (low being low difference/very similar)), first by hamming distance, then by size difference, and finally by name difference.
```

**Review by keshav-space — APPROVED** [review]
Thanks @JonoYang, LGTM!

**Review by TG1999 — APPROVED** [review]
LGTM!

**Pull request merged**